sdks/java/testing/test-utils/src/main/java/org/apache/beam/sdk/testutils/metrics/MetricsReader.java

```diff
@@ -124,9 +124,12 @@ private long getDistributionMetric(String name, DistributionType distType, long
   private <T> void checkIfMetricResultIsUnique(String name, Iterable<MetricResult<T>> metricResult)
       throws IllegalStateException {
 
+    int resultCount = Iterables.size(metricResult);
     Preconditions.checkState(
-        Iterables.size(metricResult) == 1,
-        String.format("More than one metric matches name: %s in namespace %s.", name, namespace));
+        resultCount <= 1,
+        String.format(
```

Comment:
Actually this should be just `checkState(resultCount <= 1, "..", name, namespace, resultCount)` so that it does not execute the String.format when the check passes.